yrs/src/store.rs

```diff
@@ -213,32 +176,27 @@ impl Store {
                 }
             }
         }
-        if let Some(right) = compaction.new_right {
-            if let Some(item) = self.blocks.get_item_mut(&right) {
-                item.left = Some(compaction.replacement);
-            }
+        if let Some(Block::Item(right)) = compaction.new_right.as_deref_mut() {
+            right.left = Some(compaction.replacement);
         }
     }
 
-    pub fn get_type_from_path(&self, path: &Path) -> Option<BranchRef> {
+    pub fn get_type_from_path(&self, path: &Path) -> Option<BranchPtr> {
```

Comment:
Renamed `BranchRef` &rarr; `BranchPtr` as it's a smart pointer and a value type - obeys the same rules as `BlockPtr`.